samza-azure/src/main/java/org/apache/samza/system/eventhub/EventHubSystemDescriptor.java

```diff
@@ -0,0 +1,212 @@
+   *  Larger buffer capacity typically leads to better throughput but consumes more memory.
+   *
+   * @param receiveQueueSize the number of messages from Eventhubs that should be buffered in the
+   *                      {@link org.apache.samza.util.BlockingEnvelopeMap}
+   * @return this system descriptor
+   */
+  public EventHubSystemDescriptor withReceiveQueueSize(int receiveQueueSize) {
+    this.consumerReceiveQueueSize = Optional.of(receiveQueueSize);
+    return this;
+  }
+
+  /**
+   * Maximum number of events that EventHub client can return in a receive call.
+   *
+   * @param count the number of max events per poll
+   * @return this system descriptor
+   */
+  public EventHubSystemDescriptor withMaxEventCountPerPoll(int count) {
+    this.consumerMaxEventCountPerPoll = Optional.of(count);
+    return this;
+  }
+
+  /**
+   * Number of events that EventHub client should prefetch from the server.
+   *
+   * @param count the number of events that should be prefetched.
+   * @return this system descriptor
+   */
+  public EventHubSystemDescriptor withPrefetchCount(int count) {
+    this.consumerPrefetchCount = Optional.of(count);
+    return this;
+  }
+
+
+  /**
+   * Configure the method that the message is partitioned for the downstream Eventhub in one of the following ways:
+   * <ul>
+   *   <li>ROUND_ROBIN:
+   *   The message key and partition key are ignored and the message
+   *   will be distributed in a round-robin fashion amongst all the partitions in the downstream EventHub.</li>
+   *   <li>EVENT_HUB_HASHING:
+   *   Employs the hashing mechanism in EventHubs to determine, based on the key of the message,
+   *   which partition the message should go. Using this method still ensures that all the events with
+   *   the same key are sent to the same partition in the event hub. If this option is chosen, the partition
+   *   key used for the hash should be a string. If the partition key is not set, the message key is
+   *   used instead.</li>
+   *   <li>PARTITION_KEY_AS_PARTITION:
+   *   Use the integer key specified by the partition key or key of the message to a specific partition
+   *   on Eventhub. If the integer key is greater than the number of partitions in the destination Eventhub,
+   *   a modulo operation will be performed to determine the resulting paritition.
+   *   ie. if there are 6 partitions and the key is 9, the message will end up in partition 3.
+   *   Similarly to EVENT_HUB_HASHING, if the partition key is not set the message key is used instead.</li>
+   * </ul>
+   * @param partitioningMethod
+   * @return this system descriptor
+   */
+  public EventHubSystemDescriptor withPartitioningMethod(PartitioningMethod partitioningMethod) {
+    this.producerEventhubsPartitioningMethod = Optional.of(partitioningMethod);
+    return this;
+  }
+
+  /**
+   *  Sending each message key to the eventhub in the properties of the AMQP message.
```

Comment:
Don't understand the first sentence, seems like an incomplete phrase. Can you clarify, both here and in config reference?

Reply:
Event Hubs uses eventData, which is a AMQP map under to hood. The kv-pair format from Samza is therefore not kept. Users may specify that they want to add the key in the map in addition to the value. I will clarify